### PR TITLE
Fix Presenter::argsToParams to not remove false

### PR DESCRIPTION
**Pull request by enumag**
I sometimes use boolean parameters in presenters and I want the `param=0` part in the URL because `FALSE` and `NULL` are totally different values for me. Right now however argsToParams replaces false values with nulls which breaks my use case.

### CODE REVIEW

**Comment by matej21** [comment]
:+1:

File: src/Application/UI/Presenter.php [hunk 1]
```diff
@@ -1047,7 +1047,7 @@ public static function argsToParams($class, $method, & $args, $supplemental = []
 				));
 			}
 
-			if ($args[$name] === $def || ($def === NULL && is_scalar($args[$name]) && (string) $args[$name] === '')) {
+			if ($args[$name] === $def || ($def === NULL && is_scalar($args[$name]) && !is_bool($args[$name]) && (string) $args[$name] === '')) {
```
Comment (matej21) [inline src/Application/UI/Presenter.php]: I think, `is_scalar($args[$name]) && !is_bool($args[$name]) && (string) $args[$name] === ''` can be simplified to `$args[$name] === ''` :)

Comment (enumag) [inline src/Application/UI/Presenter.php]: Probably. Integers and floats should never become an empty string after type casting.

**Comment by enumag** [comment]
@dg Thanks!

**Comment by dg** [comment]
thanks to you ;)

**Comment by dg** [comment]
The question is whether to include it in v2.3…

**Comment by enumag** [comment]
It admitedly is a BC break so as much as I'd like it in stable 2.x, putting it in 2.3 is probably not a best idea.

Are there other commits like this? If so we might want to release nette/application 2.4.

**Comment by dg** [comment]
I did not realize that it is a BC break, so it will be in 2.4.

File: src/Application/UI/Presenter.php [hunk 2]
```diff
@@ -1047,7 +1047,7 @@ public static function argsToParams($class, $method, & $args, $supplemental = []
 				));
 			}
 
-			if ($args[$name] === $def || ($def === NULL && is_scalar($args[$name]) && (string) $args[$name] === '')) {
```
Comment (Majkl578) [inline src/Application/UI/Presenter.php]: Just wondering, why did you remove is_scalar here?

**Comment by enumag** [comment]
Well since it required a change in tests and generates different URLs in some cases I would consider it a BC break...

File: src/Application/UI/Presenter.php (hunk 2 above)
Comment (enumag) [inline src/Application/UI/Presenter.php]: Because as far as I can tell `is_scalar($args[$name]) && (string) $args[$name] === ''` is equivalent to `$args[$name] === '' || $args[$name] === false`.
